src/main/java/frc/robot/swerve/SwerveConstants.java

```diff
@@ -21,27 +24,78 @@ public class SwerveConstants {
-    // maxIntegralAccum, peakOutput
-    public static final double[] FRONT_LEFT_MOTION_MAGIC_CONFIGS = {1, 0, 0, 0.2, 0, 1, 21288, 25000, 10, 5, 1};
-    public static final double[] FRONT_RIGHT_MOTION_MAGIC_CONFIGS = {1, 0, 0, 0.2, 0, 1, 21288, 25000, 10, 5, 1};
-    public static final double[] REAR_LEFT_MOTION_MAGIC_CONFIGS = {1, 0, 0, 0.2, 0, 1, 21288, 25000, 10, 5, 1};
-    public static final double[] REAR_RIGHT_MOTION_MAGIC_CONFIGS = {1, 0, 0, 0.2, 0, 1, 21288, 25000, 10, 5, 1};
-
-    public static final double[][] motionMagicConfigs = {
+    public static final Slot0Configs DRIVE_PID_GAINS =
+            new Slot0Configs()
+                    .withKP(0.005)
+                    .withKI(0.0)
+                    .withKD(0.148)
+                    .withKV(0.05)
+                    .withKS(0);
+
+    public static final Slot0Configs FRONT_LEFT_ANGLE_PID_GAINS =
+            new Slot0Configs()
+                    .withKP(1)
+                    .withKI(0)
+                    .withKD(0)
+                    .withKV(0.2)
+                    .withKS(0);
+    public static final Slot0Configs FRONT_RIGHT_ANGLE_PID_GAINS =
+            new Slot0Configs()
+                    .withKP(1)
+                    .withKI(0)
+                    .withKD(0)
+                    .withKV(0.2)
+                    .withKS(0);
+    public static final Slot0Configs REAR_LEFT_ANGLE_PID_GAINS =
+            new Slot0Configs()
+                    .withKP(1)
+                    .withKI(0)
+                    .withKD(0)
+                    .withKV(0.2)
+                    .withKS(0);
+    public static final Slot0Configs REAR_RIGHT_ANGLE_PID_GAINS =
+            new Slot0Configs()
+                    .withKP(1)
+                    .withKI(0)
+                    .withKD(0)
+                    .withKV(0.2)
+                    .withKS(0);
+
+    public static final MotionMagicConfigs FRONT_LEFT_MOTION_MAGIC_CONFIGS =
+            new MotionMagicConfigs()
+                    .withMotionMagicAcceleration(25000)
+                    .withMotionMagicCruiseVelocity(21288)
+                    .withMotionMagicJerk(1);
+    public static final MotionMagicConfigs FRONT_RIGHT_MOTION_MAGIC_CONFIGS =
+            new MotionMagicConfigs()
+                    .withMotionMagicAcceleration(25000)
+                    .withMotionMagicCruiseVelocity(21288)
+                    .withMotionMagicJerk(1);
+    public static final MotionMagicConfigs REAR_LEFT_MOTION_MAGIC_CONFIGS =
+            new MotionMagicConfigs()
+                    .withMotionMagicAcceleration(25000)
+                    .withMotionMagicCruiseVelocity(21288)
+                    .withMotionMagicJerk(1);
+    public static final MotionMagicConfigs REAR_RIGHT_MOTION_MAGIC_CONFIGS =
+            new MotionMagicConfigs()
+                    .withMotionMagicAcceleration(25000)
+                    .withMotionMagicCruiseVelocity(21288)
+                    .withMotionMagicJerk(1);
```

Comment:
Same here:
```suggestion
    public static final MotionMagicConfigs FRONT_LEFT_MOTION_MAGIC_CONFIGS =
            new MotionMagicConfigs().withMotionMagicAcceleration(25000).withMotionMagicCruiseVelocity(21288).withMotionMagicJerk(1);
    public static final MotionMagicConfigs FRONT_RIGHT_MOTION_MAGIC_CONFIGS =
            new MotionMagicConfigs().withMotionMagicAcceleration(25000).withMotionMagicCruiseVelocity(21288).withMotionMagicJerk(1);
    public static final MotionMagicConfigs REAR_LEFT_MOTION_MAGIC_CONFIGS =
            new MotionMagicConfigs().withMotionMagicAcceleration(25000).withMotionMagicCruiseVelocity(21288).withMotionMagicJerk(1);
    public static final MotionMagicConfigs REAR_RIGHT_MOTION_MAGIC_CONFIGS =
            new MotionMagicConfigs().withMotionMagicAcceleration(25000).withMotionMagicCruiseVelocity(21288).withMotionMagicJerk(1);
```
However -- do we really need a config for each module? This is not like PID gains which may be different between modules, there are just target values which shouldn't change.

Reply:
d59d580d9d0bd6c2a25dbc2efd3fb15cc8bfdbdf